crates/protocol/src/types/len_vec.rs

```diff
@@ -0,0 +1,107 @@
+//! A length-prefixed array (prefixed by a [`super::VarInt`])
+
+use bytes::{Buf, BufMut};
+
+use crate::{error, ProtocolType};
+
+/// There are 2 implementations of this:
+///
+///   - One for anything that's [`ProtocolType`] and can of be any length
+///   - One just for bytes (faster than decoding [`super::Primitive<u8>`])
+#[derive(Debug, Clone)]
+pub struct LenVec<T>(pub Vec<T>);
+
+#[derive(Debug, thiserror::Error, from_never::FromNever)]
+pub enum DecodeError<E: std::error::Error> {
+    #[error(transparent)]
+    Eof(#[from] error::Eof),
+    #[error("Failed to read LengthVec length")]
+    Length(#[from] super::varint::DecodeError),
+    #[error("Failed to read LengthVec item")]
+    Item(#[source] E),
```

Comment:
```suggestion
    #[error("Failed to read LengthVec element")]
    Element(#[source] E),
```

This seems to be a more commonly used name in `Vec`'s rustdoc and will be more distinguishable from Minecraft's items.

Reply:
Done